controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/RotationStatusUpdater.java

```diff
@@ -42,16 +44,22 @@ public RotationStatusUpdater(Controller controller, Duration interval, JobContro
     @Override
     protected void maintain() {
         var failures = new AtomicInteger(0);
+        var attempts = new AtomicInteger(0);
         var lastException = new AtomicReference<Exception>(null);
-        var applicationList = InstanceList.from(applications.asList()).hasRotation();
+        var instancesWithRotations = ApplicationList.from(applications.asList()).hasRotation().asList().stream()
+                .flatMap(application -> application.instances().values().stream())
+                .filter(instance -> ! instance.rotations().isEmpty());
 
         // Run parallel stream inside a custom ForkJoinPool so that we can control the number of threads used
         var pool = new ForkJoinPool(applicationsToUpdateInParallel);
 
         pool.submit(() -> {
-            applicationList.asList().parallelStream().forEach(application -> {
+            instancesWithRotations.parallel().forEach(instance -> {
+                attempts.incrementAndGet();
                 try {
-                    applications.lockIfPresent(application.id(), (app) -> applications.store(app.with(getStatus(app.get()))));
+                    RotationStatus status = getStatus(instance);
```

Comment:
`getStatus` moved outside of holding the application lock. Ok?

Reply:
Should be OK.